File: 349 Intersection of Two Arrays/intersection.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        
        record = set( nums1 )
        compare = set ( nums2 )
        resultSet = record & compare
        res = list(resultSet)
        return res
        

File: 349 Intersection of Two Arrays/intersection.java
public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        
        Set<Integer> record = new HashSet();
        for(int i = 0; i < nums1.length; i++)
            record.add(nums1[i]);
        
        Set<Integer> resultSet = new HashSet();
        for(int i = 0; i < nums2.length; i++)
            if( record.contains(nums2[i]) )
                resultSet.add(nums2[i]);
        
        // This time I'm using the method of converting a Set to an int Array, maybe there are other ways
        Integer[] temp = resultSet.toArray(new Integer[] {});
        int[] res = new int[temp.length];
        for(int i = 0; i < temp.length; i++)
            res[i] = temp[i].intValue();
        
        return res;
        
    }
}